src/sql/src/session/vars.rs

```diff
@@ -1739,10 +1772,25 @@ impl SystemVars {
     /// The call will return an error:
     /// 1. If `name` does not refer to a valid [`SystemVars`] field.
     pub fn reset(&mut self, name: &str) -> Result<bool, VarError> {
-        self.vars
+        let result = self
+            .vars
             .get_mut(UncasedStr::new(name))
             .ok_or_else(|| VarError::UnknownParameter(name.into()))
-            .map(|v| v.reset())
+            .map(|v| v.reset())?;
+
+        if name == MAX_CONNECTIONS.name {
+            self.refresh_state();
+        }
+        Ok(result)
+    }
+
+    /// Refresh the internal state (currently just active_connection_count) when any of
+    /// the system vars that may affect them changes. Today only when MAX_CONNECTIONS changes.
+    fn refresh_state(&mut self) {
```

Comment:
This is a bit awkward because it forces callers to check the var name before calling. What about

```
fn propagate_var_change(&mut self, name: &str)
```

which can hold the name check logic one time there. My above This would avoid the need to call this function in `new` as long as it's init'd in catalog open. Is this better or not? Hard to tell. So much shared state here it's not obvious where to put the init step. I do think catalog open is a more logical place than the new system vars though.

Reply:
I added propagate_var_change, I think that's nicer. I think doing the initilization in `SystemVars::new` makes sense b/c then you can't mess up and forget to call `refresh_state` in any code paths. I tried adding the call to Catalog::load_system_configuration but the code didn't look like it was obviously important